gzip_cache/gzip_cache.py

```diff
@@ -65,19 +65,23 @@ def create_gzip_cache(pelican):
     '''
     for dirpath, _, filenames in os.walk(pelican.settings['OUTPUT_PATH']):
         for name in filenames:
-            if should_compress(name):
+            if should_compress(name, pelican.settings):
```

Comment:
Passing the whole setting to `should_compress` seems excessive when it only cares about a single option. Would be better to extract that setting here and pass that.

Bonus: Minimal performance boost since you'll do that once instead of every file.

Reply:
I was just doing exactly the same way as the other helper function in the same file. Should I adjust both helper functions to work that way?

Reply:
I think `should_overwrite` was implemented this way to make the settings value testable. If it's evaluated in `create_gzip_cache`, its effect can no longer be observed. Should I remove the test for `should_overwrite` or leave `should_overwrite` untouched?

Reply:
Nevermind, I can keep the function but only read it once instead for each file. That will simplify things at least a bit.

Reply:
I pushed the simplification as a separate commit.

Reply:
I mean, sure it is testable but also not much point in testing `.get` method of `dict`s :).